src/lualib/New.ts

```diff
@@ -1,4 +1,29 @@
+import { __TS__Boolean } from "./Boolean";
+import { __TS__Number } from "./Number";
+import { __TS__String } from "./String";
+
+function customNewHandling(name: string, args: any[]): any {
+    switch (name) {
+        case "String":
+            return __TS__String(args[0]);
+        case "Boolean":
+            return __TS__Boolean(args[0]);
+        case "Number":
+            return __TS__Number(args[0]);
+        default:
+            return undefined;
+    }
+}
+
 export function __TS__New(this: void, target: LuaClass, ...args: any[]): any {
+    if ("name" in target && typeof target.name === "string") {
```

Comment:
The niche usecases of these functions do not warrant this performance hit on every new object being created. The choice between the different constructors can be made int he compiler at compile time, no need to burden the runtime with this.

Reply:
Concerning the classes, what way do you think we should go about handling those instead? As I did not personally find any way of directly translating `new Boolean` to `Boolean`, etc
(This comment is related to both this conversation, and the one below)